src/js/shared_controls.js

```diff
@@ -1280,6 +1280,36 @@ var SETDEX = [
 	typeof SETDEX_SV === 'undefined' ? {} : SETDEX_SV,
 ];
 
+// Creates a single dictionary for all Gen 8 Random Battles formats
+var GEN8 = {
+	"Randoms": typeof GEN8RANDOMBATTLE === 'undefined' ? {} : GEN8RANDOMBATTLE,
+	"Doubles Randoms": typeof GEN8RANDOMDOUBLESBATTLE === 'undefined' ? {} : GEN8RANDOMDOUBLESBATTLE,
+	"BDSP Randoms": typeof GEN8BDSPRANDOMBATTLE === 'undefined' ? {} : GEN8BDSPRANDOMBATTLE,
+};
+
+// COMBINED_GEN8 will be a dictionary that will have the hierarchy Pokemon -> Format -> Sets
+// An example using Absol would be:
+// {
+//		...
+//		Absol: {
+//			Randoms: {...},
+//			Doubles Randoms: {...},
+//			BDSP Randoms: {...}
+//		}
+//		...
+// }
```

Comment:
I don't think you need this huge comment for every single gen

Reply:
I mostly just copied the gen 9 code and `sed`ed 8 to 9; I can remove it